code/game/machinery/Sleeper.dm

```diff
@@ -44,6 +44,7 @@
 	available_chems = list()
 	for(var/i in 1 to I)
 		available_chems |= possible_chems[i]
+		chem_buttons |= possible_chems[i]
```

Comment:
Hey @roboticxlink 
this is spewing runtimes (check the travis details), because chem_buttons hasn't been initialized yet when doing the default parts proc (its done in reset_chem_buttons)

reset_chem_buttons should also be doing what you're trying to achieve

Can I suggest doing some to_chat debugging to find out what reset_chem_buttons isn't doing what it should be doing?

Reply:
Yeah, I can take a look.

Not sure what the travis details are though. So I'll check what's up in reset_chem_buttons

Reply:
Er, so, I just took a look with a to_chat, and it said it added them all to the list in reset_chem_buttons, and was working fine. And it doesn't look like my other change is in the one I'm using soooo... Uh, is it fixed now then? 

I'll try handupgrading it instead of just tweaking component ratings, see if that's an issue.

Reply:
Yeah, issue comes from handupgrading. Looks like it reset_chem_buttons to the first 5 twice, but the buttons still show up. Hmmmm. Is the component rating supposed to be 16? 
It looks like they go up by powers of 2 (1,2,4,8,16), and the code looks like it just looks at their rating?

But that should still be over and above, not missing them...